superset-frontend/src/dashboard/components/nativeFilters/ChartCustomization/ChartCustomizationForm.tsx

```diff
@@ -0,0 +1,1147 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { FC, useEffect, useMemo, useState, useRef, useCallback } from 'react';
+import { t, styled, css, SLOW_DEBOUNCE, useTheme } from '@superset-ui/core';
+import { debounce } from 'lodash';
+import { Form, FormItem } from 'src/components/Form';
+import { Input, TextArea } from 'src/components/Input';
+import { Radio } from 'src/components/Radio';
+import Select from 'src/components/Select/Select';
+import Collapse from 'src/components/Collapse';
+import { InfoTooltipWithTrigger } from '@superset-ui/chart-controls';
+import Loading from 'src/components/Loading';
+import { getChartDataRequest } from 'src/components/Chart/chartAction';
+import { CollapsibleControl } from '../FiltersConfigModal/FiltersConfigForm/CollapsibleControl';
+import { ColumnSelect } from '../FiltersConfigModal/FiltersConfigForm/ColumnSelect';
+import DatasetSelect from '../FiltersConfigModal/FiltersConfigForm/DatasetSelect';
+import DefaultValue from '../FiltersConfigModal/FiltersConfigForm/DefaultValue';
+import { ChartCustomizationItem } from './types';
+import { getFormData } from '../utils';
+
+const StyledForm = styled(Form)`
+  .ant-form-item {
+    margin-bottom: ${({ theme }) => theme.gridUnit * 4}px;
+  }
+`;
+
+const StyledContainer = styled.div`
+  ${({ theme }) => `
+   display: flex;
+   flex-direction: row;
+   gap: ${theme.gridUnit * 4}px;
+   padding: ${theme.gridUnit * 2}px;
+ `}
+`;
+
+const FORM_ITEM_WIDTH = 300;
+
+const StyledFormItem = styled(FormItem)`
+  width: ${FORM_ITEM_WIDTH}px;
+  margin-bottom: ${({ theme }) => theme.gridUnit * 4}px;
+
+  .ant-form-item-label > label {
+    font-size: ${({ theme }) => theme.typography.sizes.m}px;
+    font-weight: ${({ theme }) => theme.typography.weights.normal};
+    color: ${({ theme }) => theme.colors.grayscale.dark1};
+  }
+`;
+
+const CheckboxLabel = styled.span`
+  font-size: ${({ theme }) => theme.typography.sizes.m}px;
+  color: ${({ theme }) => theme.colors.grayscale.dark1};
+  font-weight: ${({ theme }) => theme.typography.weights.normal};
+`;
+
+const StyledTextArea = styled(TextArea)`
+  min-height: ${({ theme }) => theme.gridUnit * 24}px;
+  resize: vertical;
+`;
+
+const StyledRadioGroup = styled(Radio.Group)`
+  .ant-radio-wrapper {
+    font-size: ${({ theme }) => theme.typography.sizes.m}px;
+  }
+`;
+
+const StyledMarginTop = styled.div`
+  margin-top: ${({ theme }) => theme.gridUnit * 2}px;
+`;
+
+interface Props {
+  form: any;
+  item: ChartCustomizationItem;
+  onUpdate: (updatedItem: ChartCustomizationItem) => void;
+}
+
+const ChartCustomizationForm: FC<Props> = ({ form, item, onUpdate }) => {
+  const theme = useTheme();
+  const customization = useMemo(
+    () => item.customization || {},
+    [item.customization],
+  );
+
+  const [metrics, setMetrics] = useState<any[]>([]);
```

Comment:
Not going to comment on `any` as well next but all of these should be replaced with proper types